src/components/Banner/RandomBanner.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import useDelegationBanner from '../../hooks/useDelegationBanner'
+
+import useNewsletterSubscription from '../../hooks/useNewsletterSubscription'
+import { NewsletterSubscriptionModal } from '../Modal/NewsletterSubscriptionModal'
+import DelegationBanner from './Delegation/DelegationBanner'
+import SubscriptionBanner from './Subscription/SubscriptionBanner'
+
+interface Props {
+  isVisible: boolean
+}
+
+function RandomBanner({ isVisible }: Props) {
+  const {
+    showSubscriptionBanner,
+    isSubscriptionModalOpen,
+    setIsSubscriptionModalOpen,
+    onSubscriptionSuccess,
+    subscribed,
+    onClose,
+  } = useNewsletterSubscription()
+
+  const showDelegationBanner = useDelegationBanner()
+
+  if (!showSubscriptionBanner) {
+    return <DelegationBanner isVisible={isVisible} />
+  }
+
+  if (!showDelegationBanner) {
+    return (
+      <>
+        <SubscriptionBanner
+          isVisible={showSubscriptionBanner && isVisible}
+          onAction={() => setIsSubscriptionModalOpen(true)}
+        />
+        <NewsletterSubscriptionModal
+          open={isSubscriptionModalOpen}
+          onSubscriptionSuccess={onSubscriptionSuccess}
+          subscribed={subscribed}
+          onClose={onClose}
+        />
+      </>
+    )
+  }
+
+  const now = new Date()
+
+  return now.valueOf() % 2 === 0 ? (
+    <DelegationBanner isVisible={isVisible} />
+  ) : (
+    <>
```

Comment:
This components look like the ones you are returning in the other cases, maybe you can extract them to constants and do:

```
if(!B) return aComponent
if(!A) return bComponents
return random ? aComponent : bComponents
```